contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,582 @@
+    */
+    function issueExactSetFromToken(
+        ISetToken _setToken,
+        IERC20 _inputToken,
+        uint256 _amountSetToken,
+        uint256 _maxAmountInputToken,
+        ZeroExSwapQuote[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isSetToken(_setToken)
+        isWhitelistedIssuanceModule(_issuanceModule)
+        isValidInput(_setToken, _amountSetToken, _componentQuotes)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+
+        _inputToken.transferFrom(msg.sender, address(this), _maxAmountInputToken);
+        _safeApprove(_inputToken, swapTarget, _maxAmountInputToken);
+
+        uint256 totalInputTokenSold = _buyComponentsForInputToken(_setToken, _amountSetToken,  _componentQuotes, _inputToken, _issuanceModule);
+        require(totalInputTokenSold <= _maxAmountInputToken, "ExchangeIssuance: OVERSPENT TOKEN");
+
+        IBasicIssuanceModule(_issuanceModule).issue(_setToken, _amountSetToken, msg.sender);
+
+        _returnExcessInputToken(_inputToken, _maxAmountInputToken, totalInputTokenSold);
+
+        emit ExchangeIssue(msg.sender, _setToken, _inputToken, _maxAmountInputToken, _amountSetToken);
+        return totalInputTokenSold;
+    }
+
+
+    /**
+    * Issues an exact amount of SetTokens for given amount of ETH.
+    * The excess amount of tokens is returned in an equivalent amount of ether.
+    *
+    * @param _setToken              Address of the SetToken to be issued
+    * @param _amountSetToken        Amount of SetTokens to issue
+    * @param _componentQuotes       The encoded 0x transactions to execute
+    *
+    * @return amountEthReturn       Amount of ether returned to the caller
+    */
+    function issueExactSetFromETH(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        ZeroExSwapQuote[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isSetToken(_setToken)
+        isWhitelistedIssuanceModule(_issuanceModule)
+        isValidInput(_setToken, _amountSetToken, _componentQuotes)
+        external
+        nonReentrant
+        payable
+        returns (uint256)
+    {
+        require(msg.value > 0, "ExchangeIssuance: INVALID ETH AMOUNT");
+
+        IWETH(WETH).deposit{value: msg.value}();
+        _safeApprove(IERC20(WETH), swapTarget, msg.value);
+
+        uint256 totalEthSold = _buyComponentsForInputToken(_setToken, _amountSetToken, _componentQuotes, IERC20(WETH), _issuanceModule);
+
+        require(totalEthSold<= msg.value, "ExchangeIssuance: OVERSPENT ETH");
+        IBasicIssuanceModule(_issuanceModule).issue(_setToken, _amountSetToken, msg.sender);
+
+        uint256 amountEthReturn = msg.value.sub(totalEthSold);
+        if (amountEthReturn > 0) {
+            IWETH(WETH).withdraw(amountEthReturn);
+            (payable(msg.sender)).sendValue(amountEthReturn);
+        }
+
+        emit Refund(msg.sender, amountEthReturn);
```

Comment:
Actually I don't thing this refund event really makes sense to keep around anymore. In the old exchange issuance contract we added it since the refund was in ETH even when the input token was something else. Now this data feels pretty irrelevant since its just the difference between the maxInput (eth sent), and the eth used.